JdbcAggregateOperations delete by query

ghgh415263 opened the pull request:
Issue link: #1978

Add deleteAllByQuery method to JdbcAggregateOperations

This method enables deleting aggregates based on a query by performing the following steps:
1. Select root IDs matching the query with a SELECT ... FOR UPDATE to lock the rows.
2. Delete all sub-entities associated with the selected root IDs.
3. Delete the root entities identified by the selected IDs.

see
#1978

ghgh415263 commented:
I’ve opened a draft PR for #1978. If you have any suggestions for improvement or spot any issues, I’d be glad to make the necessary changes. Thank you for reviewing it!

schauder reviewed (changes requested):
Most of this looks pretty good. 
I have on major concern though: We should not transfer the ids from the locking select to the client and use it for follow up deletes.

Instead the locking select should execute on the server only, and the delete operations should be based on the query directly.

The reason for this that this operation is very attractive for large amounts of data. In such cases transporting and converting the ids creates serious load on network and memory.

And if someone prefers to select the ids first and then delete based on these ids they can still do that using multiple methods.

ghgh415263 commented:
Thanks for review

i Changed the steps as follows:
1. Lock the target rows using SELECT ... FOR UPDATE based on the query conditions.
2. Delete sub-entities by leveraging a subquery that selects the matching root rows.
3. Delete the root entities using the query conditions.

ghgh415263 committed:
JdbcAggregateOperations delete by query

Issue link: #1978

Add deleteAllByQuery method to JdbcAggregateOperations

This method enables deleting aggregates based on a query by performing the following steps:
1. Lock the target rows using SELECT ... FOR UPDATE based on the query conditions.
2. Delete sub-entities by leveraging a subquery that selects the matching root rows.
3. Delete the root entities using the query conditions.

But if the query has no criteria, deletion is performed in the same way as deleteAll method of JdbcAggregateOperations

Signed-off-by: JaeYeon Kim <ghgh415263@naver.com>

ghgh415263 commented:
hello !
I’ve resolved the conflicts and made some minor adjustments before pushing again.

schauder commented:
That's polished and merged.

Thanks.
